csrc/host_ir/pass/insert_deallocations.cpp

```diff
@@ -31,6 +43,14 @@ void InsertDeallocations::passImplementation(Fusion* fusion) {
       auto tv = val->as<TensorView>();
       last_use[tv] = i;
     }
+
+    for (auto* val : expr->outputs()) {
```

Comment:
```suggestion
    // An expression output is treated as a use to avoid premature deallocation. 
    // For example,
    // ```
    // tv = Allocate
    // tv = Matmul(...)  # at::matmul_out
    // ```
    for (auto* val : expr->outputs()) {
```
is this the reason?